core/lib/server/pub.ts

```diff
@@ -893,8 +894,21 @@ export const updatePub = async ({
 			value,
 		}));
 
+		const relevantPubFields = await getFieldInfoForSlugs({
+			slugs: vals.map(({ slug }) => slug),
+			communityId,
+			// do not update relations, and error if a relation slug is included
+			includeRelations: false,
+		});
+		const mergedPubFields = mergeSlugsWithFields(vals, relevantPubFields);
+		// Allow rich text fields to overwrite other fields
+		const { values: processedVals } = parseRichTextForPubFieldsAndRelatedPubs({
+			pubId,
+			values: mergedPubFields,
+		});
+
 		const pubValuesWithSchemaNameAndFieldId = await validatePubValues({
-			pubValues: vals,
+			pubValues: processedVals,
```

Comment:
hmm yeah i think we were thinking of two different things. I was thinking we were going to do the parsing in the `updatePub` server action here

https://github.com/pubpub/platform/blob/c72f23d8c2c0319eecb8b1dd415ea078cec20dcd/core/app/components/pubs/PubEditor/actions.ts#L48-L88

But i'm not sure i still agree with that. i was thinking that bc we update relations and pubvalues in two different places (`upsertRelations` and `updatePub`), and since the RichText field can update both, we would want to do it in a place higher up than either, eg in the server action. The server action could then call `updatePub` and `upsertRelations` respectively.

but now i'm thinking about it more i feel like that's even more clunky, as we would probably want to "update the pub while parsing the richtext value" somewhere else than just the server action.    

so maybe the following is the easiest approach:

- we rename `updatePub` to `updatePubValues`
- we create a new `updatePub` function, which does all the parsing etc
- `updatePub` calls both `updatePubValues` and `upsertPubRelations` for the relevant values.

the server action would then just call the new `updatePub`. 
this way we still have the option of being intentional about only updating a pubs values or relations, but we don't need to duplicate the parsing in many places.